boot-qemu.py

```diff
@@ -328,7 +328,10 @@ def get_linux_ver_code(decomp_cmd):
     linux_version = None
     for line in strings.stdout.decode("UTF-8").split("\n"):
         if re.search(r"Linux version \d+\.\d+\.\d+", line):
-            # "Linux version x.y.z-<localversion> ..." -> "x.y.z-<localversion>" -> "x.y.z" -> ['x', 'y', 'z']
+            # "Linux version x.y.z-<localversion> ..." OR "Linux version x.y.z+"
+            # -> "x.y.z-<localversion>" OR "x.y.z+"
+            # -> "x.y.z"
+            # -> ['x', 'y', 'z']
             linux_version = line.split(" ")[2].split("-")[0].split(
                 "+")[0].split(".")
```

Comment:
Do you think you could write a regex for this, instead of splitting on characters that might not be there?

Reply:
I think it is as simple as this...

```diff
diff --git a/boot-qemu.py b/boot-qemu.py
index 1eabd59..63b77ec 100755
--- a/boot-qemu.py
+++ b/boot-qemu.py
@@ -328,12 +328,7 @@ def get_linux_ver_code(decomp_cmd):
     linux_version = None
     for line in strings.stdout.decode("UTF-8").split("\n"):
         if re.search(r"Linux version \d+\.\d+\.\d+", line):
-            # "Linux version x.y.z-<localversion> ..." OR "Linux version x.y.z+"
-            # -> "x.y.z-<localversion>" OR "x.y.z+"
-            # -> "x.y.z"
-            # -> ['x', 'y', 'z']
-            linux_version = line.split(" ")[2].split("-")[0].split(
-                "+")[0].split(".")
+            linux_version = re.search(r"\d+\.\d+\.\d+", line)[0].split(".")
             break
     if not linux_version:
         kernel_path = decomp_cmd[-1]
```

If you would prefer that, I can update the PR.

Reply:
I ended up liking that better myself and I tested it against the few corner cases that we know of and it worked fine. PTAL.